pallets/registry/src/lib.rs

```diff
@@ -125,11 +137,21 @@ pub mod pallet {
             let old_deposit = id.deposit;
             id.deposit = T::InitialDeposit::get() + fd;
             if id.deposit > old_deposit {
-                T::Currency::reserve(&who, id.deposit - old_deposit)?;
+                T::Currency::hold(
+                    &HoldReason::RegistryIdentity.into(),
+                    &who,
+                    id.deposit - old_deposit,
+                )?;
```

Comment:
Is there a reason to do the currency -> fungible migration here? Seems unrelated to direct indexing and I think you need a migration?

Reply:
It should have gone into the 1.0 branch, but I forgot to commit it and now i'm rolling into this.

Reply:
@gregzaitsev  is this what you were referring to a while back ? I remember we had some conversation about it. Can't find the issue though